cpp_utils/src/cpp/utils.cpp

```diff
@@ -99,6 +100,55 @@ void to_uppercase(
             });
 }
 
+std::uint64_t to_bytes(
+        const std::string& input)
+{
+    static const std::map<std::string, std::uint64_t> magnitudes = {
+        {"B", 1},
+        {"KB", 1000},
+        {"MB", 1000 * 1000},
+        {"GB", 1000 * 1000 * 1000},
+        {"TB", 1000LL * 1000 * 1000 * 1000},
+        {"PB", 1000LL * 1000 * 1000 * 1000 * 1000},
+        {"KIB", 1024},
+        {"MIB", 1024 * 1024},
+        {"GIB", 1024 * 1024 * 1024},
+        {"TIB", 1024ULL * 1024 * 1024 * 1024},
+        {"PIB", 1024ULL * 1024 * 1024 * 1024 * 1024}
+    };
+
+    // Find the number and the magnitude
+    std::regex pattern(R"((\d+(?:\.\d+)?)\s*([a-zA-Z]+))");
+    std::smatch matches;
+
+    if (!std::regex_match(input, matches, pattern) || matches.size() != 3)
+    {
+        throw std::invalid_argument(
+                  "The quantity is not in the expected format. It should be a number followed by a magnitude (e.g. 10MB).");
+    }
+
+    // Extract the number
+    std::string number_str = matches[1].str();
+    double number = std::stod(number_str);
+
+    // Extract the magnitude
+    std::string magnitude_str = matches[2].str();
+    to_uppercase(magnitude_str);
+
+    const auto magnitude = magnitudes.at(magnitude_str);
+
+    // Check for overflow
+    if (number > std::numeric_limits<std::uint64_t>::max() / magnitude)
+    {
+        throw std::invalid_argument("The number is too large to be converted to bytes.");
+    }
+
+    // Calculate the number of bytes
+    const std::uint64_t bytes = number * magnitude;
```

Comment:
There is a loss of precision here as number being a double, the cast to `uint64_t` will be an approximate of the real input, although I don't know if there is something we can do about it, given that there is no `std::stoui64` 🥲. In that case, I think this should be:
```suggestion
    // Explicit cast to uint64_t is safe here as the number has already been checked to fit.
    // Furthermore, the product is also safe as the possible overflown has also been checked before
    const std::uint64_t bytes = static_cast<std::uint64_t>(number) * magnitude;
```